src/salte-auth.profile.js

```diff
@@ -0,0 +1,237 @@
+import { get, set } from 'lodash';
+import moment from 'moment';
+
+class SalteAuthProfile {
+  /**
+   * Parses the current url for the authentication values
+   */
+  constructor() {
+    if (window.salte.SalteAuthProfile.$instance) {
+      return window.salte.SalteAuthProfile.$instance;
+    }
+    window.salte.SalteAuthProfile.$instance = this;
+
+    if (location.hash) {
+      const params = location.hash.replace(/(#!?[^#]+)?#/, '').split('&');
+      for (let i = 0; i < params.length; i++) {
+        const param = params[i];
+        const [key, value] = param.split('=');
+        this.parse(key, value);
```

Comment:
In the pop-up scenario, the parameters saved to session storage will not be available to the parent window.